From d7a39ea7014db451bad6926934eeeb037627ea3d Mon Sep 17 00:00:00 2001
From: Erez Shaul <eshaul@habana.ai>
Date: Tue, 17 Sep 2024 17:39:27 +0300
Subject: [PATCH] [SW-182967] gaudi3: security: avoid use single die sram

Since SRAM_MODE_0_SINGLE_DIE_OFFSET will be removed from specs this
change will replace it with SRAM_SINGLE_DIE_OFFSET defined in
address_space.h

Change-Id: I87bec1009a1cc9ba7d85e8e103937f6a0b07171e
Signed-off-by: Erez Shaul <eshaul@habana.ai>
---

diff --git a/arch/gaudi3/common/security.c b/arch/gaudi3/common/security.c
index 35a4290..95bf847 100644
--- a/arch/gaudi3/common/security.c
+++ b/arch/gaudi3/common/security.c
@@ -18,6 +18,7 @@
 #include <mstr_if_rr_hbw_regs.h>
 #include <gaudi3_security.h>
 #include <gaudi3/gaudi3.h>
+#include <gaudi3/address_space.h>
 #include <gaudi3_efuse.h>
 #include <hl_sec_if.h>
 #include <hl_fw_if.h>
@@ -1064,7 +1065,7 @@
 		return;
 	}
 
-	image_base_full = (SRAM_BASE_ADDR + SRAM_MODE_0_SINGLE_DIE_OFFSET)
+	image_base_full = (SRAM_BASE_ADDR + SRAM_SINGLE_DIE_OFFSET)
 						+ (mdesc.base & IMAGE_OFFSET_MASK);
 
 	/*
